substrate/primitives/blockchain/src/backend.rs

```diff
@@ -229,10 +228,37 @@ pub trait Backend<Block: BlockT>:
 	) -> std::result::Result<DisplacedLeavesAfterFinalization<Block>, Error> {
 		let mut result = DisplacedLeavesAfterFinalization::default();
 
-		if finalized_block_number == Zero::zero() {
+		let leaves = self.leaves()?;
+
+		// If we have only one leaf there are no forks, and we can return early.
+		if finalized_block_number == Zero::zero() || leaves.len() == 1 {
 			return Ok(result)
 		}
 
+		let first_leaf = leaves[0];
```

Comment:
nit: `let Some(first_leaf) = leaves.get(0) else {..}` since this may panic if leaves is empty (although I think its unlikely)